projects/angular/src/forms/common/if-control-state/if-control-state.service.ts

```diff
@@ -34,11 +35,26 @@ export class IfControlStateService implements OnDestroy {
           // Subscribe to the status change events, only after touched
           // and emit the control
           this.subscriptions.push(
-            this.control.statusChanges.subscribe(() => {
+            this.control.statusChanges?.subscribe(() => {
```

Comment:
Don't we need to subscribe to ngControlService.controlChanges here too, same as we do with the additional changes below?
We're only subscribing to statusChanges of the primary control, but we're not reestablishing the status subscriptions if control gets changed.

Reply:
'ngControlService.controlChanges' already there in line#32. Code is already available, request you to check complete file.